pwndbg/arch.py

```diff
@@ -26,9 +26,15 @@
 
 
 def fix_arch(arch):
-    arches = ['x86-64', 'i386', 'mips', 'powerpc', 'sparc', 'arm', 'aarch64', arch]
-    return next(a for a in arches if a in arch)
+    for match in ['x86-64', 'i386', 'mips', 'powerpc', 'sparc', 'aarch64']:
+        if match in arch:
+            return match
 
+    # Distinguish between Cortex-M and other ARM
+    if 'arm' in arch:
+        return 'armcm' if arch.endswith('-m') else 'arm'
```

Comment:
I'm not totally sure how watertight this check is, but I think this should work.

Reply:
So I'm not entirely sure, but I think that https://github.com/bminor/binutils-gdb/blob/502c64b9ac12cf2a35d3cb55c51e2eefd33a2494/bfd/cpu-arm.c#L223 is where gdb gets its architecture info from. 
Correct me if I'm wrong, but it looks like the cortex-m's would be:
```
  N (bfd_mach_arm_6M,        "armv6-m",        FALSE, & arch_info_struct[20]),
  N (bfd_mach_arm_6SM,       "armv6s-m",       FALSE, & arch_info_struct[21]),
  N (bfd_mach_arm_7EM,       "armv7e-m",       FALSE, & arch_info_struct[22]),
  N (bfd_mach_arm_8M_BASE,   "armv8-m.base",   FALSE, & arch_info_struct[25]),
  N (bfd_mach_arm_8M_MAIN,   "armv8-m.main",   FALSE, & arch_info_struct[26]),

```
If so, it looks like it might make more sense to do `return 'armcm' if '-m' in arch else 'arm'`

Reply:
You're right. I'll push a fixed commit.